backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -202,6 +219,50 @@ public int cancelAll() {
 	  return _repo.cancelAllPendingOrders(_os.getCurrentOrganization());
   }
 
+    @Transactional
+    @AuthorizationConfiguration.RequirePermissionUpdateTest
+    public TestEvent correctTestMarkAsError(UUID testEventId, String reasonForCorrection) {
+        // The client sends us a TestEvent, we need to map back to the Order.
+        Optional<TestEvent> loadExistingEvent = _terepo.findById(testEventId);
+        if (loadExistingEvent.isEmpty()) {
+            // should this throw?
+            LOG.error("Failed to load TestEvent by id {}", testEventId);
+            return null;
+        }
+        TestEvent event = loadExistingEvent.get();
+        if (event.getCorrectionStatus() == TestCorrectionStatus.REMOVED) {
+            throw new IllegalGraphqlArgumentException("Can not correct removed test event");
+        }
+
+        // todo: should we verify reasonForCorrection is NOT empty? Do we trim()?
+
+        TestOrder order = _repo.findByTestEventId(event.getOrganization(), testEventId);
+        if (order == null) {
+            throw new IllegalGraphqlArgumentException("TestEvent: could not load the parent order");
+        }
+
+        // generate a duplicate test_event that just has a status of REMOVED and the reason
+        TestEvent newRemoveEvent = new TestEvent(event, TestCorrectionStatus.REMOVED, reasonForCorrection);
+        _terepo.save(newRemoveEvent);
+
+        // order having reason text is way more useful when we allow actual corrections not just deletes.
+        order.setReasonForCorrection(reasonForCorrection);
+        order.setTestEventRef(newRemoveEvent);
+
+        // order.setOrderStatus(OrderStatus.CANCELED); NO: this makes it disappear.
+
+        // We currently don't do anything special with this CorrectionStatus on the order, but in the next
+        // refactor, we will set this to TestCorrectionStatus.CORRECTED and it will go back into the queue to
+        // be corrected.
+        order.setCorrectionStatus(TestCorrectionStatus.REMOVED);
+
+        // NOTE: WHEN we support actual corrections (versus just deleting). Make sure to fix the
+        // TestOrder.dateTestedBackdate field, it may not be NULL and will override the correction date!
```

Comment:
I am not sure what the desired treatment of date fields here is, and this comment makes me less sure.